services/MerchStoreService.ts

```diff
@@ -264,27 +262,20 @@ export default class MerchStoreService {
     return order.getPublicOrder();
   }
 
-  public async getAllOrders(user: UserModel, canSeeAllOrders = false): Promise<PublicOrder[]> {
+  public async getAllOrders(user: UserModel, status?: OrderStatus, canSeeAllOrders = false): Promise<PublicOrder[]> {
```

Comment:
Let's split this up into separate `getAllOrdersForUser` and `getAllOrdersForAllUsers` methods and then call the appropriate one from the controller (and do `getPublicOrder` in the controller).